filter.go

```diff
@@ -0,0 +1,90 @@
+package inflight
+
+import (
+	"net/http"
+	"strconv"
+	"sync"
+	"time"
+)
+
+type FQFilter struct {
+	// list-watching API models
+	lock   *sync.Mutex
+	queues []*Queue
+
+	*sharedDispatcher
+
+	Matcher  func(*http.Request, []*Queue) *Queue
+	Delegate http.HandlerFunc
+}
+
+func (f *FQFilter) Serve(resp http.ResponseWriter, req *http.Request) {
+
+	matchedQueue := f.Matcher(req, f.queues)
+	// fmt.Printf("matchedQueue: %d\n", matchedQueue.Index)
+
```

Comment:
I see that no locks are held at this point.  Whatever `f.Matcher` looked at while making its decision could change at this point.